drivers/frequency/adf5902/adf5902.c

```diff
@@ -0,0 +1,940 @@
+			    ADF5902_REG0_PLO(ADF5902_POWER_UP_LO) |
+			    ADF5902_REG0_PTX1(ADF5902_POWER_DOWN_TX1) |
+			    ADF5902_REG0_PTX2(ADF5902_POWER_UP_TX2) |
+			    ADF5902_REG0_PADC(ADF5902_POWER_UP_ADC) |
+			    ADF5902_REG0_VCAL(ADF5902_VCO_NORMAL_OP) |
+			    ADF5902_REG0_PVCO(ADF5902_POWER_UP_VCO) |
+			    ADF5902_REG0_TX1C(ADF5902_TX1_NORMAL_OP) |
+			    ADF5902_REG0_TX2C(ADF5902_TX2_AMP_CAL));
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Required 500us delay */
+	udelay(500);
+
+	return ret;
+}
+
+/**
+ * @brief VCO Normal Operation
+ * @param dev - The device structure.
+ * @return Returns 0 in case of success or negative error code.
+ */
+static int32_t adf5902_vco_normal_op(struct adf5902_dev *dev)
+{
+	int32_t ret;
+
+	/* VCO Normal Operation */
+	ret = adf5902_write(dev, ADF5902_REG9 | ADF5902_REG9_RESERVED_NORMAL);
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Compute new f_pfd */
+	ret = adf5902_vco_freq_param(dev);
+	if (ret != SUCCESS)
+		return ret;
+
+	ret = adf5902_write(dev, ADF5902_REG7 | ADF5902_REG7_RESERVED |
+			    ADF5902_REG7_R_DIVIDER(dev->ref_div_factor) |
+			    ADF5902_REG7_REF_DOUBLER(dev->ref_doubler_en) |
+			    ADF5902_REG7_R_DIV_2(ADF5902_R_DIV_2_DISABLE) |
+			    ADF5902_REG7_CLK_DIV(dev->clk1_div_ramp) |
+			    ADF5902_REG7_MASTER_RESET(ADF5902_MASTER_RESET_DISABLE));
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Set LSB FRAC */
+	ret = adf5902_write(dev, ADF5902_REG6 | ADF5902_REG6_RESERVED |
+			    ADF5902_REG6_FRAC_LSB_WORD(dev->frac_lsb));
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Set MSB FRAC and INT */
+	ret = adf5902_write(dev, ADF5902_REG5 | ADF5902_REG5_RESERVED |
+			    ADF5902_REG5_FRAC_MSB_WORD(dev->frac_msb) |
+			    ADF5902_REG5_INTEGER_WORD(dev->int_div));
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Set Ramp Status */
+	ret = adf5902_write(dev, ADF5902_REG4 | ADF5902_REG4_RESERVED |
+			    ADF5902_REG4_TEST_BUS(dev->ramp_status));
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Set I/O Voltage to 3.3V */
+	ret = adf5902_write(dev, ADF5902_REG3 | ADF5902_REG3_RESERVED |
+			    ADF5902_REG3_READBACK_CTRL(ADF5902_REG_RB_NONE) |
+			    ADF5902_REG3_IO_LVL(ADF5902_IO_LVL_3V3) |
+			    ADF5902_REG3_MUXOUT(ADF5902_MUXOUT_RAMP_STATUS));
+	if (ret != SUCCESS)
+		return ret;
+
+	udelay(100);
+
+	/* Set Ramp Mode */
+	ret = adf5902_write(dev, ADF5902_REG11 | ADF5902_REG11_RESERVED |
+			    ADF5902_REG11_RAMP_MODE(dev->ramp_mode));
+	if (ret != SUCCESS)
+		return ret;
+
+	return ret;
+}
+/**
+ * @brief Initializes the ADF5902.
+ * @param device - The device structure.
+ * @param init_param - The structure containing the device initial parameters.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int32_t adf5902_init(struct adf5902_dev **device,
+		     struct adf5902_init_param *init_param)
+{
+	int32_t ret;
+	uint32_t i;
+	struct adf5902_dev *dev;
+
+	/* Check ADF5902 Initialization Paramteres */
+	ret = adf5902_check_init_param(init_param);
+	if(ret != SUCCESS)
+		goto error;
```

Comment:
You can just return ret

Reply:
if the unconditional jump is already implemented in this function, I would use it as much as I can and avoid mixing it with direct returns.

Reply:
Actually, I will remove the `error:` completely I `return ret;` :)